### PR TITLE
Fix world creation

### PR DESCRIPTION
**Pull request by sunny-viktoryia**
fixes https://github.com/sparkletown/internal-sparkle-issues/issues/1767

Reasoning: we need to check if the world slug exists in both cases:
* when we create a world (not having world Id yet) and check if the slug is already taken by another world
* when we edit a world and we want the generated slug to exist only in the current world

### CODE REVIEW

File: src/utils/validation.ts [hunk 1]
```diff
@@ -32,31 +30,21 @@ export const testVenueByNameExists = async (value: unknown) => {
...
     .collection("worlds")
     .where("slug", "==", slug)
-    .where("isHidden", "==", false)
-    .where(firebase.firestore.FieldPath.documentId(), "!=", worldId)
-    .get();
+    .where("isHidden", "==", false);
+
+  if (worldId) {
+    path = path.where(firebase.firestore.FieldPath.documentId(), "!=", worldId);
+  }
```
Comment (alexey-petrov) [inline src/utils/validation.ts:45]: maybe we should be using something like this?
`
const path = firebase .firestore().collection("worlds").where("slug", "==", slug).where("isHidden", "==", false);
const result = worldId ? path.where(firebase.firestore.FieldPath.documentId(), "!=", worldId) : path;
`
Just so we avoid this unnecessary `path` mutation

Feel free to rename `result` into something else, I'm bad at naming :)

**Review by alexxpetrov — COMMENTED** [review]
One question

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
@sunny-viktoryia please check the suggestions and give us input on what you think is feasable. I don't think we'll too much time to check if this fix will catch all scenarios

File: src/utils/validation.ts [hunk 2]
```diff
@@ -32,31 +30,21 @@ export const testVenueByNameExists = async (value: unknown) => {
...
-    throw new SparkleFetchError({
-      message: `Invalid worldId: ${String(worldId)} for value: ${String(
-        value
-      )}`,
-      args: {
-        worldId,
-        value,
-      },
-    });
-  }
```
[the remark this answers is not in the document]
Comment (sunny-viktoryia) [reply src/utils/validation.ts:47]: but when `worldId` is `undefined` it means that we're creating a world, and I don't thing any error should be raised. Why should I bring this piece back?.. 😕 

<hr>

if you go lower in the code you'll find that world id serves for filtering found slugs. If slug exist and the world id its world id is the same as the one of the world we're editing right now, we ignore it. 

but if we're creating a world, we don't have to filter out anything.

[the remark this answers is not in the document]
Comment (goran-peoski-work) [reply src/utils/validation.ts:47]: Yeah, forgot about that. Didn't put 2 and 2 together

**Review by goran-peoski-work — APPROVED** [review]
I don't think the questions we posed might be blockers (a refactor nice to have, not essential atm), and this PR could be merged as long as the logic works for 3 cases:
1. world is being created
2. world slug is being changed (with or without other fields)
3. world is being updated, but slug not changed

File: src/utils/validation.ts (hunk 1 above)
Comment (sunny-viktoryia) [reply src/utils/validation.ts:62]: @alexey-petrov, done!
https://github.com/sparkletown/sparkle/pull/2936/commits/28189cf95ab26774051c2503b83fd0f499fda1ba

File: src/utils/validation.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sunny-viktoryia) [reply src/utils/validation.ts:47]: moved that error catching lower:
https://github.com/sparkletown/sparkle/pull/2936/commits/28189cf95ab26774051c2503b83fd0f499fda1ba

**Comment by sunny-viktoryia** [comment]
@goran-peoski-work, I checked all 3 scenario.
While checking duplicated slugs we don't count a world by world id (if exists). 
so if
1. world is creating, there is nothing to filter out and we check if the generated world slug exists in any world
2. world slug is changed. We've filtered out our current world (comparing by world id which should be never changed) and check if the new slug exists in any other world.
3. world slug is not changed. Again, we filter out our current world and taking into account that world slugs should be unique it should work ok.

Also, good to think is that ok that we also filter out where `world.isHidden = false`. Then if someone just changes the field in the firestore to `true`, we might end up with 2 worlds with the same world slug, is that a bug?

**Pull request merged**